include/pagmo/algorithms/ipopt.hpp

```diff
@@ -0,0 +1,1179 @@
+ * the optimised individual will be inserted back into the population.
+ * The selection and replacement strategies can be configured via set_selection(const std::string &),
+ * set_selection(population::size_type), set_replacement(const std::string &) and
+ * set_replacement(population::size_type).
+ *
+ * Configuring the optimsation run
+ * -------------------------------
+ *
+ * Ipopt supports a large amount of options for the configuration of the optimisation run. The options
+ * are divided into three categories:
+ * - *string* options (i.e., the type of the option is ``std::string``),
+ * - *integer* options (i.e., the type of the option is ``Ipopt::Index`` - an alias for some integer type, typically
+ *   ``int``),
+ * - *numeric* options (i.e., the type of the option is ``double``).
+ *
+ * The full list of options is available on the
+ * <a href="https://www.coin-or.org/Ipopt/documentation/node40.html">Ipopt website</a>. pagmo::ipopt allows to configure
+ * any Ipopt option via methods such as ipopt::set_string_options(), ipopt::set_string_option(),
+ * ipopt::set_integer_options(), etc., which need to be used before invoking ipopt::evolve().
+ *
+ * If the user does not set any option, pagmo::ipopt will use Ipopt's default values for the options (see the
+ * <a href="https://www.coin-or.org/Ipopt/documentation/node40.html">documentation</a>), with the following
+ * modifications:
+ * - if the ``"print_level"`` integer option is **not** set by the user, it will be set to 0 by pagmo::ipopt (this will
+ *   suppress most screen output produced by the solver - note that we support an alternative form of logging via
+ *   the ipopt::set_verbosity() method);
+ * - if the ``"hessian_approximation"`` string option is **not** set by the user and the optimisation problem does
+ *   **not** provide the Hessians, then the option will be set to ``"limited-memory"`` by pagmo::ipopt. This makes it
+ *   possible to optimise problems without Hessians out-of-the-box (i.e., Ipopt will approximate numerically the
+ *   Hessians for you);
+ * - if the ``"constr_viol_tol"`` numeric option is **not** set by the user and the optimisation problem is constrained,
+ *   then pagmo::ipopt will compute the minimum value ``min_tol`` in the vector returned by pagmo::problem::get_c_tol()
+ *   for the optimisation problem at hand. If ``min_tol`` is nonzero, then the ``"constr_viol_tol"`` Ipopt option will
+ *   be set to ``min_tol``, otherwise the default Ipopt value (1E-4) will be used for the option. This ensures that,
+ *   if the constraint tolerance is not explicitly set by the user, a solution deemed feasible by Ipopt is also
+ *   deemed feasible by pagmo (but the opposite is not necessarily true).
+ *
+ * \verbatim embed:rst:leading-asterisk
+ * .. warning::
+ *
+ *    A moved-from pagmo::ipopt is destructible and assignable. Any other operation will result
+ *    in undefined behaviour.
+ *
+ * .. note::
```

Comment:
At one point I realised it's possible to embed sphinx syntax into doxygen with the construct above. It's a bit more cumbersome and, of course, looks like shit in the doxygen output, but renders much better in sphinx.